tests/integration/telemetry/stackdriver/common.go

```diff
@@ -252,22 +267,148 @@ func ConditionallySetupMetadataServer(ctx resource.Context) (err error) {
 	return nil
 }
 
-func ValidateLogs(srvLogEntry, clName, trustDomain string, filter stackdriver.LogType) error {
+func ValidateLogs(t test.Failer, srvLogEntry, clName, trustDomain string, filter stackdriver.LogType) error {
 	var wantLog loggingpb.LogEntry
 	if err := unmarshalFromTemplateFile(srvLogEntry, &wantLog, clName, trustDomain); err != nil {
 		return fmt.Errorf("logs: failed to parse wanted log entry: %v", err)
 	}
+	return ValidateLogEntry(t, &wantLog, filter)
+}
 
+func ValidateLogEntry(t test.Failer, want *loggingpb.LogEntry, filter stackdriver.LogType) error {
 	// Traverse all log entries received and compare with expected server log entry.
 	entries, err := SDInst.ListLogEntries(filter, EchoNsInst.Name())
 	if err != nil {
 		return fmt.Errorf("logs: failed to get received log entries: %v", err)
 	}
 
 	for _, l := range entries {
-		if proto.Equal(l, &wantLog) {
+		l.Trace = ""
+		l.SpanId = ""
+		if proto.Equal(l, want) {
 			return nil
 		}
 	}
-	return fmt.Errorf("logs: did not get expected log entry: got %v\n want %v", entries, wantLog.String())
+	LogAccessLogsDiff(t, want, entries)
+	return fmt.Errorf("logs: did not get expected log entry")
+}
+
+func LogAccessLogsDiff(t test.Failer, wantRaw *loggingpb.LogEntry, entries []*loggingpb.LogEntry) {
+	query := normalizeLogs(wantRaw)
+	existing := []map[string]string{}
+	for _, e := range entries {
+		existing = append(existing, normalizeLogs(e))
+	}
+	logDiff(t, query, existing)
+}
+
+func LogTraceDiff(t test.Failer, wantRaw *cloudtrace.Trace, entries []*cloudtrace.Trace) {
+	query := normalizeTrace(wantRaw)
+	existing := []map[string]string{}
+	for _, e := range entries {
+		existing = append(existing, normalizeTrace(e))
+	}
+	logDiff(t, query, existing)
+}
+
+func LogMetricsDiff(t test.Failer, wantRaw *monitoring.TimeSeries, entries []*monitoring.TimeSeries) {
+	query := normalizeMetrics(wantRaw)
+	existing := []map[string]string{}
+	for _, e := range entries {
+		existing = append(existing, normalizeMetrics(e))
+	}
+	logDiff(t, query, existing)
+}
+
+func logDiff(t test.Failer, query map[string]string, entries []map[string]string) {
+	if len(entries) == 0 {
+		t.Logf("no entries found")
```

Comment:
Will we be able to tell which types of entries are not found?

Reply:
no , let me add that